packages/typescript-estree/src/convert.ts

```diff
@@ -297,83 +327,106 @@ export default function convert(config: ConvertConfig): ESTreeNode | null {
-   * @returns {Object}    the converted ESTree name object
+   * @param tagName the tagName object from a JSX ts.Node
+   * @param parent
+   * @returns the converted ESTree name object
    */
-  function convertTypeScriptJSXTagNameToESTreeName(
-    tagName: ts.JsxTagNameExpression
-  ): ESTreeNode {
-    const tagNameToken = convertToken(tagName, ast);
-
-    if (tagNameToken.type === AST_NODE_TYPES.JSXMemberExpression) {
+  protected convertTypeScriptJSXTagNameToESTreeName(
+    tagName: ts.JsxTagNameExpression,
+    parent: ts.Node
+  ): es.JSXMemberExpression | es.JSXIdentifier {
+    // TODO: remove convertToken call
+    const tagNameToken = convertToken(tagName, this.ast);
+
+    if (tagName.kind === SyntaxKind.PropertyAccessExpression) {
       const isNestedMemberExpression =
-        (node as any).tagName.expression.kind ===
-        SyntaxKind.PropertyAccessExpression;
+        tagName.expression.kind === SyntaxKind.PropertyAccessExpression;
 
       // Convert TSNode left and right objects into ESTreeNode object
       // and property objects
-      tagNameToken.object = convertChild((node as any).tagName.expression);
-      tagNameToken.property = convertChild((node as any).tagName.name);
+      const object = this.convertChild(tagName.expression, parent);
+      const property = this.convertChild(tagName.name, parent);
 
       // Assign the appropriate types
-      tagNameToken.object.type = isNestedMemberExpression
+      object.type = isNestedMemberExpression
         ? AST_NODE_TYPES.JSXMemberExpression
         : AST_NODE_TYPES.JSXIdentifier;
-      tagNameToken.property.type = AST_NODE_TYPES.JSXIdentifier;
+      property.type = AST_NODE_TYPES.JSXIdentifier;
       if ((tagName as any).expression.kind === SyntaxKind.ThisKeyword) {
-        tagNameToken.object.name = 'this';
+        object.name = 'this';
       }
+
+      return this.createNode<es.JSXMemberExpression>(tagName, {
+        type: AST_NODE_TYPES.JSXMemberExpression,
+        range: tagNameToken.range,
+        loc: tagNameToken.loc,
+        object: object,
+        property: property
+      });
     } else {
-      tagNameToken.type = AST_NODE_TYPES.JSXIdentifier;
-      tagNameToken.name = tagNameToken.value;
+      return this.createNode<es.JSXIdentifier>(tagName, {
+        type: AST_NODE_TYPES.JSXIdentifier,
+        range: tagNameToken.range,
+        loc: tagNameToken.loc,
+        name: tagNameToken.value
+      });
     }
-
-    delete tagNameToken.value;
-
-    return tagNameToken;
   }
 
   /**
    * Applies the given TS modifiers to the given result object.
-   * @param {ts.ModifiersArray} modifiers original ts.Nodes from the node.modifiers array
-   * @returns {void} (the current result object will be mutated)
+   * @param result
+   * @param modifiers original ts.Nodes from the node.modifiers array
+   * @returns the current result object will be mutated
+   * @deprecated
+   * TODO: remove this
```

Comment:
Can you expand on this comment?

```
TODO: remove this
```

Is an action that could be completed at any time, I'm guessing something has to happen before that is appropriate? Right now it is unclear

Reply:
in all nodes except 2 we are handing unsupported modifers in diferent way than for `es.TSEnumDeclaration | es.TSModuleDeclaration`, i should write appropriate message here

Reply:
```
* @deprecated This function adds not standardized `modifiers` property in nodes
```